ql/time/calendar.hpp

```diff
@@ -241,8 +242,8 @@ namespace QuantLib {
         return (d.month() != adjust(d+1).month());
     }
 
-    inline Date Calendar::endOfMonth(const Date& d) const {
-        return adjust(Date::endOfMonth(d), Preceding);
+    inline Date Calendar::endOfMonth(const Date& d, BusinessDayConvention conv) const {
+        return adjust(Date::endOfMonth(d), conv);
```

Comment:
This is supposed to return the end of month according to a given calendar; it doesn't make sense for this to return the first day of the next month if we're passing `Following`.